File: factories/llm_service_factory.py
from services.openai_service import OpenAIService
from services.ollama_service import OllamaService

class LLMServiceFactory:
    @staticmethod
    def get_service(provider: str):
        if provider.lower() == 'openai':
            return OpenAIService()
        elif provider.lower() == 'ollama':
            return OllamaService()
        else:
            raise ValueError("Unsupported provider")

File: src/factories/LLMServiceFactory.ts
// src/services/LLMServiceFactory.ts
import { ILLMService } from "../interfaces/ILLMService";
import { OpenAIService } from "../services/OpenAIService";

export class LLMServiceFactory {
    static getService(provider: string): ILLMService {
        switch (provider.toLowerCase()) {
            case "openai":
                return new OpenAIService();
            default:
                throw new Error("Unsupported provider");
        }
    }
}
